src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_LIST_SIZE = 10;
+    private int size;
+    private Object[] list;
+
+    public ArrayList() {
+        list = new Object[DEFAULT_LIST_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == list.length) {
+            list = increase(list);
+        }
+        list[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == list.length) {
+            list = increase(list);
+        }
+        if ((index >= 0) && (index <= size)) {
+            Object[] temporary = new Object[list.length];
+            System.arraycopy(list, 0, temporary,0, index);
+            temporary[index] = value;
+            System.arraycopy(list, index, temporary,index + 1, size - index);
+            list = temporary;
```

Comment:
Let's try to use `arraycopy` method only once, you can avoid using temporary array

Reply:
I've done it by explanation of Bogdan Chupika:
https://youtu.be/YhSF8_hVIU4?t=74
As I understand, you want that I changed 29-33 to:
System.arraycopy(list, index, list,index + 1, size - index);
list[index] = value;
Right?